config-example.yaml

```diff
@@ -95,6 +95,16 @@ derp:
     # For more details on how this works, check this great article: https://tailscale.com/blog/how-tailscale-works/
     stun_listen_addr: "0.0.0.0:3478"
 
+    # For better connection stability (especially when using an Exit-Node and DNS is not working), 
+    # it is possible to optionall add the public IPv4 and IPv6 address to the Derp-Map using:
+    IPv4: 1.2.3.4
+    IPv6: 2001:db8::1
```

Comment:
```suggestion
    ipv4: 1.2.3.4
    ipv6: 2001:db8::1
```
 or maybe
```suggestion
    ip:
      v4: 1.2.3.4
      v6: 2001:db8::1
```

Reply:
What happens if only one is set? do we need to require both?

Reply:
It would be pretty cool if there was an option to resolve this on startup, but, on the other hand, that is kind of a can of worms on how to handle changes, how to resolve etc.

Reply:
I would say having them as two seperate options is prettier.
If none is set we have the behaviour we have right now. If one is set it is just using IPv4 or IPv6, if both are set the client can choose.


I thought about getting them dynamically too, but that could be a problem, when the IP changes via DNS. I do not know if the DERP-Map is actually cached in Headscale and what happens if it is overridden with an internal IP on Headscale itself. So I thought better to simply add them to the config. 

It could be done dynamically though like in bootstrap-DNS: https://github.com/alexhalbi/headscale/blob/6e734bc864abc13f8d608cb7e566611bf9593b2f/hscontrol/derp/server/derp_server.go#L228
```go
addrs, err := resolver.LookupIP(resolvCtx, "ip", node.HostName)
```
It could be added here: https://github.com/alexhalbi/headscale/blob/6e734bc864abc13f8d608cb7e566611bf9593b2f/hscontrol/derp/server/derp_server.go#L86

Reply:
Sure, we dont have to do automatic resolving, but the option should be renamed the the latter example I think.